src/freenet/client/filter/M3UFilter.java

```diff
@@ -0,0 +1,190 @@
+                } else {
+                    isComment = false;
+                }
+                // read one line as a fileUri
+                numberOfDotsInUri = 0;
+                isBadUri = false;
+                fileIndex = 0;
+                fileUri = new byte[MAX_URI_LENGTH];
+                while (readcount != -1) {
+                    if (!isComment && 
+                        // do not include carriage return in filenames
+                        !Arrays.equals(nextbyte, CHAR_CARRIAGE_RETURN) &&
+                        // enforce maximum path length to avoid OOM attacks
+                        fileIndex <= MAX_URI_LENGTH) {
+                        fileUri[fileIndex] = nextbyte[0];
+                        fileIndex += readcount;
+                    }
+                    readcount = dis.read(nextbyte);
+                    if (Arrays.equals(nextbyte, CHAR_NEWLINE)) {
+                        if (!isComment) {
+                            // remove too long paths
+                            if (fileIndex <= MAX_URI_LENGTH) {
+                                for (int i = 0; i < fileIndex; i++) {
+                                    byte b = fileUri[i];
+                                    if (!isAllowedInUri(b)) {
+                                        isBadUri = true;
+                                        break;
+                                    } 
+                                    if ((byte)'.' == b) {
+                                        numberOfDotsInUri += 1;
+                                    }
+                                }
+                                if (numberOfDotsInUri != 1) {
+                                    isBadUri = true;
+                                }
+                                // use only the first fileIndex bytes
+                                String uri;
+                                // TODO: cleanly filter the URI, i.e. with processURI from GenericReadFilterCallback.
+                                if (isBadUri) {
+                                    uri = badUriReplacement;
+                                } else {
+                                    uri = new String(fileUri, 0, fileIndex, "UTF-8");
+                                    uri += uriForcetypeMp3Suffix;
+                                }
+                                output.write(uri.getBytes("UTF-8"));
+                                output.write(nextbyte);
+                            }
+                        }
+                        // skip the newline
+                        readcount = dis.read(nextbyte);
+                        break;
+                    }
+                }
+            }
+        }
+        output.flush();
+    }
+    
+    private static String l10n(String key) {
+        return NodeL10n.getBase().getString("M3UFilter."+key);
+    }
+    
+    private void throwHeaderError(String shortReason, String reason) throws DataFilterException {
```

Comment:
The PR has to be evaluated on its own merit, I can't merge stuff based on the promise that it might be useful in another one... especially when I'm trying to reduce the amount of code we maintain.